src/main/java/nextstep/courses/domain/session/enrollment/FreeEnrollment.java

```diff
@@ -6,16 +6,18 @@
 
 public class FreeEnrollment implements Enrollment {
     private final Status status;
+    private final EnrollmentStatus enrollmentStatus;
     private final Students students;
 
-    public FreeEnrollment(Status status, Students students) {
+    public FreeEnrollment(Status status, EnrollmentStatus enrollmentStatus, Students students) {
         this.status = status;
+        this.enrollmentStatus = enrollmentStatus;
         this.students = students;
     }
 
     @Override
     public void enroll(NsUser student, Payment payment) {
-        if (!status.isRecruit()) {
+        if (!status.isRecruit() || enrollmentStatus.isImPossible()) {
```

Comment:
모집 상태는 분리되었으니 진행 상태에 맞는 이름으로 변경하면 어떨까요?
```suggestion
        if (!status.진행중() || enrollmentStatus.isImPossible()) {
```